lms/djangoapps/course_blocks/transformers/tests/test_load_override_data.py

```diff
@@ -63,14 +71,40 @@ def test_transform(self, field):
         OverrideDataTransformer.collect(self.block_structure)
 
         # transform phase
-        OverrideDataTransformer().transform(
+        OverrideDataTransformer(self.learner).transform(
             usage_info=self.course_usage_key,
             block_structure=self.block_structure,
         )
 
         # verify overridden data
         assert get_override_for_user(self.learner, self.block, field) == expected_overrides.get(field)
 
+    @ddt.data(*REQUESTED_FIELDS)
+    def test_transform_for_selected_learner(self, field):
+        """
+        assert that data transformed for selected learner
+        """
+        override_field_for_user(
+            self.learner,
+            self.block,
+            field,
+            expected_overrides.get(field)
+        )
+
+        # collect phase
+        OverrideDataTransformer.collect(self.block_structure)
+
+        # transform phase
+        OverrideDataTransformer(self.learner).transform(
+            usage_info=self.course_usage_key,
+            block_structure=self.block_structure,
+        )
+
+        # verify learner has overridden data
+        assert get_override_for_user(self.learner, self.block, field) == expected_overrides.get(field)
+        # other learner2 dont have overridden data
+        assert get_override_for_user(self.learner2, self.block, field) is None
```

Comment:
I'm not totally familiar with block transformers to completely understand this, but there are no other internal comments so here is what I have to offer for now:

Based on my understanding of the field override infrastructure, get_override_for_user checks only what override_field_for_user sets (i.e. these are basically setters and getters on `StudentFieldOverride` objects with caching), so it seems to me that the effect of the above "transform phase" is not actually tested.  Does the transformer have any side-effects at the DB level, or does it only affect an in-memory block structure?  In any case, would it be prudent to additionally examine the transformed block structure?

Reply:
thanks for you feedback @pwnage101. I will look into this.

Reply:
@pwnage101

1) we are just loading data from `StudentFieldOverride`.
2)  Once it is loaded we are setting block data in `override_xblock_field`. block_data is a in memory object. Which we have until it is destroyed from memory.
https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/content/block_structure/block_structure.py#L404
3) We have not saving back anything to database.

I have this answers you questions.
Thanks